### PR TITLE
[Analyzer] Allow calling #generate_version_locking_dependencies witho…

### PR DESCRIPTION
**Pull request by segiddins**
…ut a sandbox set up

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* [Analyzer] Allow calling #generate_version_locking_dependencies witho…  
  [segiddins](https://github.com/segiddins)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/installer/analyzer.rb [hunk 1]
```diff
@@ -185,6 +185,10 @@ def update_mode
         end
       end
 
+      def podfile_dependencies
```
Comment (dnkoutso) [inline lib/cocoapods/installer/analyzer.rb]: is this public? does it it need to be?

File: lib/cocoapods/installer/analyzer.rb [hunk 2]
```diff
@@ -722,9 +726,10 @@ def generate_version_locking_dependencies(podfile_state)
         else
           deleted_and_changed = podfile_state.changed + podfile_state.deleted
           deleted_and_changed += pods_to_update[:pods] if update_mode == :selected
-          local_pod_names = @podfile_dependency_cache.podfile_dependencies.select(&:local?).map(&:root_name)
+          local_pod_names = podfile_dependencies.select(&:local?).map(&:root_name)
           pods_to_unlock = local_pod_names.to_set.delete_if do |pod_name|
-            sandbox.specification(pod_name).checksum == lockfile.checksum(pod_name)
+            next unless sandbox_specification = sandbox.specification(pod_name)
```
Comment (dnkoutso) [inline lib/cocoapods/installer/analyzer.rb]: does this fix something else?

Comment (dnkoutso) [reply lib/cocoapods/installer/analyzer.rb]: oh nm I think this can fix a crasher.

Comment (segiddins) [reply lib/cocoapods/installer/analyzer.rb]: yeah, this is the actual fix, the other change is just to allow testing it

File: lib/cocoapods/installer/analyzer.rb (hunk 1 above)
Comment (segiddins) [reply lib/cocoapods/installer/analyzer.rb]: nope, it's private and can stay that way

**Pull request merged**